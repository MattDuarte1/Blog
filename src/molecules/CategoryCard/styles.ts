import styled, { css, DefaultTheme } from 'styled-components';

const MediaContainer = {
  small: (theme: DefaultTheme) => css`
    @media ${theme.media.lteSmall} {
      width: 80%;
    }
  `,
};

interface ContainerProps {
  isActive: boolean;
}

const MenuIsActive = (isActive: boolean, theme: DefaultTheme) =>
  isActive &&
  css`
    background-color: ${theme.colors.blue};
    color: #fff;
  `;

export const Container = styled.article<ContainerProps>`
  ${({ theme, isActive }) => css`
    display: flex;
    align-items: center;
    padding-left: ${theme.spacings.large};
    width: 220px;
    height: 280px;
    color: ${theme.colors.black};
    background-color: ${theme.colors.white};
    filter: drop-shadow(4px 6px 13px rgba(0, 0, 0, 0.1));
    border-radius: 8px;
    cursor: pointer;

    > div {
      display: flex;
      flex-direction: column;
      align-items: flex-start;
      text-align: center;
      padding: 0px;
      gap: 30px;
    }

    &:hover {
      background-color: ${theme.colors.blue};
      color: white;
    }

    ${MediaContainer.small(theme)}
    ${MenuIsActive(isActive, theme)}
  `}
`;
