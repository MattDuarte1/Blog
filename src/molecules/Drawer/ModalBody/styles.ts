import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    flex: 1 1 0%;
    display: flex;
    padding: 2rem;
    width: 100%;
    height: inherit;
    justify-content: flex-start;
    gap: 2rem;
    flex-direction: column;
    align-items: center;

    > a {
      min-width: 100%;

      &:hover {
        color: ${theme.colors.blue};
      }
    }
  `}
`;
