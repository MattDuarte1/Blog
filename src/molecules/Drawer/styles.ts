import styled, { css } from 'styled-components';

interface IModalProps {
  isOpen?: boolean;
}

export const ModalBackDrop = styled.div<IModalProps>`
  ${({ isOpen }) => css`
    position: fixed;
    right: 0px;
    top: 0px;
    width: 100vw;
    transform: translateX(${isOpen ? '0%' : '100%'});
    height: 100vh;
    background-color: rgba(0, 0, 0, 0.48);
    z-index: 100;
    transition: all 0.8s ease-in-out;
  `}
`;

export const Modal = styled.div<IModalProps>`
  ${({ theme, isOpen }) => css`
    position: fixed;
    right: 1px;
    top: 0px;
    bottom: 0px;
    width: ${isOpen ? '285px' : '0px'};
    padding: 25px;
    height: inherit;
    background-color: ${theme.colors.white};
    z-index: 110;
    transition: all 1s ease;

    > svg {
      position: absolute;
      top: 10px;
      right: 2rem;

      font-size: ${theme.font.sizes.large};
      color: ${theme.colors.black};
      cursor: pointer;
      transition: all 0.5s ease;

      &:hover {
        color: ${theme.colors.blue};
        transform: scale(1.1);
      }
    }
  `}
`;
