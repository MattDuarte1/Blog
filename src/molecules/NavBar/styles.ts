import styled, { css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`
    display: flex;
    height: 44px;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 0px;
    gap: 50px;

    @media ${theme.media.lteLarge} {
      display: none;
    }
  `}
`;
