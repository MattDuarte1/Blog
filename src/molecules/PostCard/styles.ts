import styled, { css, DefaultTheme } from 'styled-components';

const MediaBreapoints = {
  AuthorDescription: {
    small: (theme: DefaultTheme) => css`
      @media ${theme.media.lteSmall} {
        > img {
          width: 47px;
          height: 47px;
        }
      }
    `,
  },
};

export const Container = styled.article`
  ${({ theme }) => css`
    width: 300px;
    height: 425px;
    padding: 1.4rem;
    background: ${theme.colors.white};
    position: relative;

    box-shadow: 4px 6px 13px rgba(0, 0, 0, 0.1);
    border-radius: 12px;
    text-align: start;
    transition: all 0.2s ease;
    cursor: pointer;

    > h1 {
      margin-left: 1rem;
    }

    img {
      margin-bottom: 2rem;
    }

    &:hover {
      transform: scale(1.02);
      filter: drop-shadow(0 1.5px 2px rgba(0, 0, 0, 0.2));
    }
  `}
`;

export const AuthorContent = styled.div`
  ${({ theme }) => css`
    position: absolute;
    padding: 0 1.4rem;
    width: 100%;
    height: 57px;
    left: 0px;
    top: 348px;
    display: flex;
    color: ${theme.colors.black};
  `}
`;

export const AuthorDescription = styled.div`
  ${({ theme }) => css`
    display: flex;
    width: 90%;
    flex-direction: row;
    justify-content: center;
    padding: 0px;
    gap: 15px;
    color: ${theme.colors.black};

    ${MediaBreapoints.AuthorDescription['small'](theme)}
  `}
`;

export const Avatar = styled.div`
  overflow: hidden;
  border-radius: 50%;
  width: 50px;
  height: 50px;
`;

export const AuthorInfo = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 2px;
  gap: 4.2px;

  width: 184px;
  height: 46.2px;
`;

export const ViewBox = styled.div`
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
  flex-flow: column nowrap;

  > svg {
    color: ${({ theme }) => theme.colors.black};
    font-size: ${({ theme }) => theme.font.sizes.medium};
  }

  > h1 {
    display: inline;
  }
`;
