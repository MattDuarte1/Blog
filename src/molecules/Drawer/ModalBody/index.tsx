import { IMenu } from '@/interfaces/Menu';
import Menu from '@/atoms/Menu';
import SubscriptionButton from '@/atoms/SubscriptionButton';
import { useSession, signIn, signOut } from 'next-auth/react';
import * as Styled from './styles';
import { forwardRef } from 'react';

type ModalBodyProps = {
  data: readonly IMenu[];
};

const ModalBody = forwardRef<HTMLDivElement, ModalBodyProps>(
  ({ data, ...props }, ref) => {
    const { data: session } = useSession();
    return (
      <Styled.Container ref={ref} {...props}>
        <SubscriptionButton
          session={session}
          onSubscribe={() => signIn('github')}
          onUnSubscribe={() => signOut()}
        />
        {data.map((menu, index) => (
          <Menu
            key={index}
            title={menu.title}
            link={menu.href}
            newTab={menu.newtab}
          />
        ))}
      </Styled.Container>
    );
  },
);

ModalBody.displayName = 'ModalBody Component';

export default ModalBody;
