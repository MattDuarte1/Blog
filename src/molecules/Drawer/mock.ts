import { theme } from '@/styles/theme';

export default {
  state: {
    drawerIsOpen: false,
    theme: theme,
  },
  menus: [
    { title: 'Inicio', newtab: false, href: '/#header' },
    { title: 'Sobre', newtab: false, href: '/#hero' },
    { title: 'Categoria', newtab: false, href: '/#categoria' },
    { title: 'Posts', newtab: false, href: '/#posts' },
  ],
} as const;
