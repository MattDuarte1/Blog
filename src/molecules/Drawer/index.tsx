import { forwardRef, MutableRefObject } from 'react';
import * as Styled from './styles';
import ModalBody from './ModalBody';
import { IoMdClose } from 'react-icons/io';

import MockDrawer from './mock';

type DrawerProps = {
  drawerIsOpen: boolean;
  activeDrawer: () => void;
};

const Drawer = forwardRef<HTMLDivElement, DrawerProps>(
  (
    { drawerIsOpen, activeDrawer, ...props },
    ref: MutableRefObject<HTMLDivElement>,
  ) => {
    const handleBackDropClick = (e: React.SyntheticEvent) => {
      const target = e.target as HTMLDivElement | HTMLAnchorElement;
      if (
        target.getAttribute('aria-label') === 'modal-backdrop' ||
        target.getAttribute('target')
      ) {
        activeDrawer();
      }
    };

    return (
      <Styled.ModalBackDrop
        aria-label="modal-backdrop"
        aria-expanded={drawerIsOpen}
        ref={ref}
        onClick={handleBackDropClick}
        isOpen={drawerIsOpen}
        {...props}
      >
        <Styled.Modal
          role="dialog"
          aria-describedby="-modal-"
          isOpen={drawerIsOpen}
        >
          <IoMdClose aria-label="close-button" onClick={activeDrawer} />
          <ModalBody aria-describedby="modal-body" data={MockDrawer.menus} />
        </Styled.Modal>
      </Styled.ModalBackDrop>
    );
  },
);

Drawer.displayName = 'Drawer Component';

export default Drawer;
