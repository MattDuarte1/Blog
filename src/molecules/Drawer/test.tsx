import { fireEvent, screen } from '@testing-library/react';
import Drawer from '.';
import { renderTheme } from '@/styles/render-theme';
import { useSession } from 'next-auth/react';
import React from 'react';

jest.mock('next-auth/react');
const mockUserSession = useSession as jest.Mock;
mockUserSession.mockReturnValue({
  status: 'unauthenticated',
  data: null,
});

const activeDrawer = jest.fn();

describe('<Drawer>', () => {
  it('Should render correctly Drawer component with modal not active', () => {
    renderTheme(<Drawer drawerIsOpen={false} activeDrawer={activeDrawer} />);
    const backDropModal = screen.getByLabelText('modal-backdrop');

    expect(backDropModal).toBeInTheDocument();
    expect(backDropModal).toHaveAttribute('aria-expanded', `false`);
    expect(backDropModal).toHaveStyle(`
      transform: translateX(100%);
      background: rgba(0, 0, 0, 0.48);
      position: fixed;
    `);
  });

  it('Should render correctly Drawer component with modal active', () => {
    renderTheme(<Drawer drawerIsOpen={true} activeDrawer={activeDrawer} />);
    const backDropModal = screen.getByLabelText('modal-backdrop');
    const Modal = screen.getByRole('dialog');

    expect(backDropModal).toHaveAttribute('aria-expanded', 'true');
    expect(Modal).toBeInTheDocument();
    expect(backDropModal).toHaveStyle(`
      transform: translateX(0%);
      background-color: rgba(0, 0, 0, 0.48);
      position: fixed;
    `);
  });

  it('Should call function when clicked in backdropModal', () => {
    renderTheme(<Drawer drawerIsOpen={true} activeDrawer={activeDrawer} />);
    const backDropModal = screen.getByLabelText('modal-backdrop');

    expect(backDropModal).toBeInTheDocument();
    expect(backDropModal.getAttribute('aria-expanded')).toBe('true');

    fireEvent.click(backDropModal);
    expect(activeDrawer).toHaveBeenCalled();
  });

  it('Should call function when clicked in close Button', () => {
    renderTheme(<Drawer drawerIsOpen={true} activeDrawer={activeDrawer} />);
    const closeButton = screen.getByLabelText('close-button');

    expect(closeButton).toBeInTheDocument();

    fireEvent.click(closeButton);
    expect(activeDrawer).toHaveBeenCalled();
  });

  it('Should call function when clicked in menus', () => {
    renderTheme(<Drawer drawerIsOpen={true} activeDrawer={activeDrawer} />);
    const menuInicio = screen.getByRole('link', { name: /inicio/i });

    expect(menuInicio).toBeInTheDocument();
    fireEvent.click(menuInicio);
    expect(activeDrawer).toHaveBeenCalled();
  });

  it('Should match snapshot drawer open', () => {
    const { container } = renderTheme(
      <Drawer drawerIsOpen={true} activeDrawer={activeDrawer} />,
    );

    expect(container).toMatchSnapshot();
  });
});
