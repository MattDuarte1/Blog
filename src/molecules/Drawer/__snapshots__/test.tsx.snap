// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer> Should match snapshot drawer open 1`] = `
.c0 {
  position: fixed;
  right: 0px;
  top: 0px;
  width: 100vw;
  -webkit-transform: translateX(0%);
  -ms-transform: translateX(0%);
  transform: translateX(0%);
  height: 100vh;
  background-color: rgba(0,0,0,0.48);
  z-index: 100;
  -webkit-transition: all 0.8s ease-in-out;
  transition: all 0.8s ease-in-out;
}

.c1 {
  position: fixed;
  right: 1px;
  top: 0px;
  bottom: 0px;
  width: 285px;
  padding: 25px;
  height: inherit;
  background-color: #FFFFFE;
  z-index: 110;
  -webkit-transition: all 1s ease;
  transition: all 1s ease;
}

.c1 > svg {
  position: absolute;
  top: 10px;
  right: 2rem;
  font-size: 3.576rem;
  color: #2B2C34;
  cursor: pointer;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
}

.c1 > svg:hover {
  color: #6246EA;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c5 {
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 600;
  font-size: 1.617rem;
  line-height: 25px;
  text-transform: capitalize;
  color: #2B2C34;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
  max-width: 100px;
  height: 25px;
}

.c3 {
  background: #6246EA;
  border: none;
  border-radius: 25px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.9rem;
  padding: 1rem;
  cursor: pointer;
}

.c4 {
  font-family: 'Plus Jakarta Sans','sans-serif';
  font-style: normal;
  font-weight: 700;
  line-height: 1.2;
  text-transform: capitalize;
  font-size: 1.617rem;
  text-transform: none;
  color: #2B2C34;
}

.c2 {
  -webkit-flex: 1 1 0%;
  -ms-flex: 1 1 0%;
  flex: 1 1 0%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 2rem;
  width: 100%;
  height: inherit;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 2rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 > a {
  min-width: 100%;
}

.c2 > a:hover {
  color: #6246EA;
}

<div>
  <div
    aria-expanded="true"
    aria-label="modal-backdrop"
    class="c0"
  >
    <div
      aria-describedby="-modal-"
      class="c1"
      role="dialog"
    >
      <svg
        aria-label="close-button"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M405 136.798L375.202 107 256 226.202 136.798 107 107 136.798 226.202 256 107 375.202 136.798 405 256 285.798 375.202 405 405 375.202 285.798 256z"
        />
      </svg>
      <div
        aria-describedby="modal-body"
        class="c2"
      >
        <button
          class="c3"
        >
          <svg
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 480 512"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M186.1 328.7c0 20.9-10.9 55.1-36.7 55.1s-36.7-34.2-36.7-55.1 10.9-55.1 36.7-55.1 36.7 34.2 36.7 55.1zM480 278.2c0 31.9-3.2 65.7-17.5 95-37.9 76.6-142.1 74.8-216.7 74.8-75.8 0-186.2 2.7-225.6-74.8-14.6-29-20.2-63.1-20.2-95 0-41.9 13.9-81.5 41.5-113.6-5.2-15.8-7.7-32.4-7.7-48.8 0-21.5 4.9-32.3 14.6-51.8 45.3 0 74.3 9 108.8 36 29-6.9 58.8-10 88.7-10 27 0 54.2 2.9 80.4 9.2 34-26.7 63-35.2 107.8-35.2 9.8 19.5 14.6 30.3 14.6 51.8 0 16.4-2.6 32.7-7.7 48.2 27.5 32.4 39 72.3 39 114.2zm-64.3 50.5c0-43.9-26.7-82.6-73.5-82.6-18.9 0-37 3.4-56 6-14.9 2.3-29.8 3.2-45.1 3.2-15.2 0-30.1-.9-45.1-3.2-18.7-2.6-37-6-56-6-46.8 0-73.5 38.7-73.5 82.6 0 87.8 80.4 101.3 150.4 101.3h48.2c70.3 0 150.6-13.4 150.6-101.3zm-82.6-55.1c-25.8 0-36.7 34.2-36.7 55.1s10.9 55.1 36.7 55.1 36.7-34.2 36.7-55.1-10.9-55.1-36.7-55.1z"
            />
          </svg>
          <h1
            class="c4"
          >
            Conectar com Github
          </h1>
        </button>
        <a
          class="c5"
          href="/#header"
          target="_self"
        >
          Inicio
        </a>
        <a
          class="c5"
          href="/#hero"
          target="_self"
        >
          Sobre
        </a>
        <a
          class="c5"
          href="/#categoria"
          target="_self"
        >
          Categoria
        </a>
        <a
          class="c5"
          href="/#posts"
          target="_self"
        >
          Posts
        </a>
      </div>
    </div>
  </div>
</div>
`;
